lib/puppet/provider/package/gem.rb

```diff
@@ -108,6 +108,15 @@ def install(useversion = true)
     self.fail "Could not install: #{output.chomp}" if output.include?("ERROR")
   end
 
+  # Allow Gem version expressions in ensure
+  def insync?(is)
+    begin
+      Gem::Dependency.new('', resource[:ensure]).match?('', is)
+    rescue
+      false
```

Comment:
Note: this is necessary in case `is` is not a valid version number (e.g. "absent").